agent/logging.go

```diff
@@ -87,33 +96,99 @@ func (a *Agent) makeLoggerProducer(ctx context.Context, c *model.LoggerConfig, t
 			LogkeeperURL:      a.opts.LogkeeperURL,
 			LogkeeperBuilder:  task.Id,
 			LogkeeperBuildNum: task.Execution,
-			Sender:            model.LogSender(agentConfig.Type),
+			Sender:            agentConfig.Type,
 			SplunkServerURL:   agentConfig.SplunkServer,
 			SplunkToken:       agentConfig.SplunkToken,
-			Filepath:          fmt.Sprintf("%s/agent.log", path),
+			Filepath:          fmt.Sprintf("%s/%s", a.opts.WorkingDirectory, agentLogs),
 		})
 	}
 	for _, systemConfig := range c.System {
 		config.System = append(config.System, client.LogOpts{
 			LogkeeperURL:      a.opts.LogkeeperURL,
 			LogkeeperBuilder:  task.Id,
 			LogkeeperBuildNum: task.Execution,
-			Sender:            model.LogSender(systemConfig.Type),
+			Sender:            systemConfig.Type,
 			SplunkServerURL:   systemConfig.SplunkServer,
 			SplunkToken:       systemConfig.SplunkToken,
-			Filepath:          fmt.Sprintf("%s/system.log", path),
+			Filepath:          fmt.Sprintf("%s/%s", a.opts.WorkingDirectory, systemLogs),
 		})
 	}
 	for _, taskConfig := range c.Task {
 		config.Task = append(config.Task, client.LogOpts{
 			LogkeeperURL:      a.opts.LogkeeperURL,
 			LogkeeperBuilder:  task.Id,
 			LogkeeperBuildNum: task.Execution,
-			Sender:            model.LogSender(taskConfig.Type),
+			Sender:            taskConfig.Type,
 			SplunkServerURL:   taskConfig.SplunkServer,
 			SplunkToken:       taskConfig.SplunkToken,
-			Filepath:          fmt.Sprintf("%s/task.log", path),
+			Filepath:          fmt.Sprintf("%s/%s", a.opts.WorkingDirectory, taskLogs),
 		})
 	}
 	return a.comm.GetLoggerProducer(ctx, td, &config)
 }
+
+type s3UploadOpts struct {
+	bucket    string
+	awsKey    string
+	awsSecret string
+}
+
+func (a *Agent) uploadToS3(ctx context.Context, tc *taskContext, opts s3UploadOpts) error {
+	s3Opts := pail.S3Options{
+		Credentials: pail.CreateAWSCredentials(opts.awsKey, opts.awsSecret, ""),
+		Region:      endpoints.UsEast1RegionID,
+		Name:        opts.bucket,
+		Permission:  "public-read",
+		ContentType: "text/plain",
+	}
+	bucket, err := pail.NewS3Bucket(s3Opts)
+	if err != nil {
+		return errors.Wrap(err, "error creating pail")
+	}
+
+	return a.uploadLogFiles(ctx, tc, bucket, opts.bucket)
+}
+
+func (a *Agent) uploadLogFiles(ctx context.Context, tc *taskContext, bucket pail.Bucket, name string) error {
+	agentLogFile, err := os.Open(fmt.Sprintf("%s/%s", a.opts.WorkingDirectory, agentLogs))
+	if err != nil {
+		if !os.IsNotExist(err) {
+			grip.Error(errors.Wrap(err, "error opening agent log file"))
+		}
```

Comment:
maybe we can refactor this pattern to use some helper to check existance before opening the file? 

also the Bucket Interface has upload/download helpers which will read local files without needing to do os.Open

Reply:
refactored so it's less verbose